tests/test_polaraveraging.py

```diff
@@ -0,0 +1,225 @@
+"""
+Tests for polaraveraging.py
+"""
+import clmm
+import clmm.polaraveraging as pa
+from numpy import testing
+import numpy as np
+from astropy.table import Table
+import os
+
+def test_make_bins():
+    testing.assert_equal(len( pa.make_bins(1,10,9,False)),10 )
+    testing.assert_allclose( pa.make_bins(1,10,9,False) , np.arange(1.,11.) )
+    testing.assert_allclose( pa.make_bins(1,10000,4,True) ,10.**(np.arange(5)) )
+    
+    testing.assert_raises(TypeError, pa.make_bins, rmin='glue', rmax=10, n_bins=9, log_bins=False)
+    testing.assert_raises(TypeError, pa.make_bins, rmin=1, rmax='glue', n_bins=9, log_bins=False)
+    testing.assert_raises(TypeError, pa.make_bins, rmin=1, rmax=10, n_bins='glue', log_bins=False)
+    testing.assert_raises(TypeError, pa.make_bins, rmin=1, rmax=10, n_bins=9, log_bins='glue')
+
+    testing.assert_raises(ValueError, pa.make_bins, rmin=1, rmax=10, n_bins=-4, log_bins=False)
+    testing.assert_raises(ValueError, pa.make_bins, rmin=1, rmax=-10, n_bins=9, log_bins=False)
+    testing.assert_raises(ValueError, pa.make_bins, rmin=1, rmax=10, n_bins=0, log_bins=False)
+    testing.assert_raises(TypeError, pa.make_bins, rmin=1, rmax=10, n_bins=9.9, log_bins=False)
+
+def test_compute_g_x():
+    data = np.array([[0.01, 0.02, 0.01], # g1
+                     [0.01, 0.02, 0.03], # g2
+                     [0.1, 1.2, 3.]]) # phi [rad]
+
+    # test that function works for scalar and vector input
+    assert(isinstance(pa._compute_g_x(*data[:,0]), float))
```

Comment:
For this block of tests, call func once and save it to var. Then check that var however you want. A little cleaner than recalling the function every time.